homeassistant/components/yeelight/__init__.py

```diff
@@ -279,21 +281,19 @@ def _update_properties(self):
 
         return self._available
 
-    def _initialize_if_not_already(self):
-        if self._initialized:
-            return
-
+    def _initialize_device(self):
         self._initialized = True
         dispatcher_send(self._hass, DEVICE_INITIALIZED, self.ipaddr)
 
     def update(self):
         """Update device properties and send data updated signal."""
-        if self._update_properties():
-            self._initialize_if_not_already()
-
+        self._update_properties()
         dispatcher_send(self._hass, DATA_UPDATED.format(self._ipaddr))
 
     def setup(self):
-        """Initialize device and send initialized signal."""
-        if self._update_properties() or self.model:
-            self._initialize_if_not_already()
+        """Fetch initial device properties."""
+        initial_update = self._update_properties()
+
+        if not initial_update and self.model:
+            """We can build correct class anyway."""
```

Comment:
Use single line comment (`#`) here, other than that this looks good to go from my side :-)

Reply:
Right, even lint complained about it. Fixed